__tests__/app.test.js

```diff
@@ -66,19 +66,56 @@ describe("NCNews API testing", () => {
       return request(app)
         .get("/api/articles")
         .expect(200)
-        .then(({ body: { topics } }) => {
-          expect(topics).toBeSorted({ descending: true });
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSorted({ descending: true });
         });
     });
     test("comment count should be different according to the total comments in any given article ", () => {
       return request(app)
         .get("/api/articles")
         .expect(200)
-        .then(({ body: { topics } }) => {
-          expect(topics[5].comment_count).toBe("11");
-          expect(topics[topics.length - 1].comment_count).toBe("0");
+        .then(({ body: { articles } }) => {
+          expect(articles[5].comment_count).toBe("11");
+          expect(articles[articles.length - 1].comment_count).toBe("0");
         });
     });
+    describe("QUERY/api/articles", () => {
+      test("return filter the topic by specified query in this case...only cat topics allowed! ", () => {
+        return request(app)
+          .get("/api/articles?topic=cats")
+          .expect(200)
+          .then((response) => {
+            const articles = response.body.articles;
+            articles.forEach((article) => {
+              expect(article.topic).toBe("cats");
+            });
+          });
+      });
+      test.only("sort articles by article id in descending order  ", () => {
+        return request(app)
+          .get("/api/articles?sort_by=article_id")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            expect(articles).toBeSortedBy("article_id", { descending: true });
+          });
+      });
+      test("be able to specify an article order i.e ascending or descending ", () => {
+        return request(app)
+          .get("/api/articles?sort_by=article_id&&order=asc")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            expect(articles).toBeSortedBy("article_id", { descending: false });
+          });
```

Comment:
would benefit from a test to see that sort and order work alongside category filter